ui/page/wallets/wallet_page.go

```diff
@@ -1079,6 +1085,60 @@ func (pg *WalletPage) deleteBadWallet(badWalletID int) {
 		}).Show()
 }
 
+func (pg *WalletPage) listenForTxNotifications() {
+	go func() {
+		for {
+			var notification interface{}
+
+			select {
+			case notification = <-pg.Receiver.NotificationsUpdate:
+			case <-pg.ctx.Done():
+				return
+			}
+
+			switch n := notification.(type) {
+			case wallet.NewTransaction:
+				// refresh wallets when new transaction is received
+				pg.updateAccountBalance()
+				pg.RefreshWindow()
+			case wallet.SyncStatusUpdate:
+				// refresh wallet account and balance on every new block
+				// only if sync is completed.
+				if n.Stage == wallet.BlockAttached && pg.WL.MultiWallet.IsSynced() {
+					pg.updateAccountBalance()
+					pg.RefreshWindow()
+				}
+			}
+		}
+	}()
+}
+
+func (pg *WalletPage) updateAccountBalance() {
+	pg.listLock.Lock()
+	listItems := pg.listItems
+	pg.listLock.Unlock()
+
+	for _, item := range listItems {
+		wal := pg.WL.MultiWallet.WalletWithID(item.wal.ID)
+		if wal != nil {
+			accountsResult, err := wal.GetAccountsRaw()
+			if err != nil {
+				continue
+			}
+
+			var totalBalance int64
+			for _, acc := range accountsResult.Acc {
+				totalBalance += acc.TotalBalance
+			}
+
+			pg.listLock.Lock()
+			item.totalBalance = dcrutil.Amount(totalBalance).String()
+			item.accounts = accountsResult.Acc
+			pg.listLock.Unlock()
```

Comment:
Does this method take long to complete? Why not `pg.listLock.Lock()` and `defer pg.listLock.Unlock()` at the top of the method, then range through `pg.listItems` and make modifications as necessary.